_i18n/en/resources/user-guides/cli_wallet_daemon_isolation_qubes_whonix.md

```diff
@@ -1,25 +1,122 @@
 
-+ Create a `systemd` file.
+### Before installation
+
++ Go to the template settings and change net qube from none to sys-whonix (remember to change it back to none after installation is completed!).
+
+### Installation
+
++ Open a terminal in the template.
+
++ (Optional) update the system.
 
 ```
-user@host:~$ sudo nano /home/user/monerod.service
+[template workstation user ~]% sudo apt update && sudo apt upgrade
+```
+
++ Install wget (if not already installed)
+
+```
+[template workstation user ~]% sudo apt install wget
+```
+
++ Download the latest Monero release.
+
+```
+[template workstation user ~]% wget 'https://downloads.getmonero.org/gui/linux64'
+```
+
++ Extract the archive.
+
+```
+[template workstation user ~]% tar xf linux64
+```
+
++ Install the binaries (name of the Monero folder might be different for you depending on the current Monero version).
+
+```
+[template workstation user ~]% cd monero-gui-v0.18.3.3
+[template workstation user ~/monero-gui-v0.18.3.3]% sudo install extras/* monerod monero-wallet-gui /bin
+```
+
+### (Optional if you intend to use CLI wallet) Configure desktop entry for Monero GUI and icon
+
++ Create a desktop entry for Monero GUI wallet.
+
+```
+[template workstation user ~]% sudo nano /usr/share/applications/monero.desktop                              
+```
+
+Paste the following contents:
+
+```
+[Desktop Entry]
+Name=Monero GUI
+Exec=monero-wallet-gui
+Type=Application
+Terminal=false
+Categories=Network;Qt;Finance;Office;
+Icon=/usr/share/icons/monero.png
+StartupNotify=true
+```
+
++ Download Monero icon and move to the appropriate location
+
+```
+[template workstation user ~]% wget https://github.com/monero-project/monero-gui/blob/master/images/appicons/256x256.png
+[template workstation user ~]% sudo mv 256x256.png /usr/share/icons/monero.png
+```
+
+### After installation
+
++ (Optional) Remove installation files
+
+```
+[template workstation user ~]% rm linux64                 
+[template workstation user ~]% rm -rf monero-gui-v0.18.3.3 
+```
+
++ Shutdown the template
+
++ Go to the template settings and change net qube back to none.
+
+### Updating Monero
+
+If you want to update Monero - simply follow all the steps listed in this section again, except the section about configuring the desktop entry and icon.
+
+## 2. Create `monero-wallet-ws` and `monerod-ws` AppVMs:
+
+We need to create two AppVMs to isolate the wallet from the daemon.
+
+* Create `monero-wallet-ws` AppVM that will be used for Monero wallet using whonix-workstation as the template and set networking to none.
+
+* Create `monerod-ws` AppVM that will be used for Monero daemon using whonix-workstation as the template and set networking to sys-whonix. Before moving on, make sure that this workstation has enough private storage. You can estimate how much space you need by checking the size of the [raw blockchain]({{ site.baseurl }}/downloads/#blockchain). Keep in mind that the blockchain will take up more space with time.
```

Comment:
The message about checking the size of the raw blockchain is outdated (the file is 86G while actual blockchain is 200G).
It should either be replaced with a hardcoded number (e.g. "make sure that this AppVM has at least 250 gigabytes of private storage") which is not ideal because it will quickly get outdated, or with another source of truth for current blockchain size.

Reply:
If recommending a pruned node, the blockchain sizendoesnt change as quickly. 

I personally recommend an abundance of free space to account for wear and future.
If someone asks me, i say 128-256gb for pruned node, 512-1tb for full node, and that its better to go big. (write cycles run out slower on a larger drive, and the blockchain grows at rates that cant be predicted)

Reply:
I'm afraid that writing something like "it's recommended to have 512gb-1tb of private storage" might dissuade some/many people from running a full node because this number is too high.

And just to make sure that we are on the same page - in Qubes, "private storage" refers to how much space is allocated to a Qube (a VM), not to the total capacity of your SSD.
![Screenshot_2024-07-28_11-59-46](https://github.com/user-attachments/assets/81f8728c-be86-44be-ba6a-57bd5c89d4c2)
Also due to the way AppVMs work (only /home is persistent), size of files other than the blockchain can be neglected (what I'm trying to say - we don't need to account for "everything else" when writing how much "private storage" a user needs, only blockchain size should be taken into account).
For example, on my system (we ignore everything that is not in /home, as it is not persisted anyway):
```
[workstation user ~]% du -s ~           
209266636	/home/user
[workstation user ~]% du -s ~/.bitmonero
209265992	/home/user/.bitmonero
```
So the total space taken by the Qube is almost the same as size of the blockchain :) 

And also, this "private storage" number can be increased at any time. You can start with 250G, and change it to 300G as soon as you get a notification that your qube has not enough space.

Reply:
I think it might be good idea to mention in the guide that running a pruned node is also an option.

I've never tried that though, is simply adding `--prune-blockchain` to the daemon start params enough?

Reply:
I guess if we don't have a good resource that shows the current size of blockchain (btw I still don't understand why https://moneroj.net/blockchainsize/ shows current blockchain size as ~147G, while my synced blockchain seems to be 200G in size), the best option is to write something like "at the time of writing this guide, the recommended minimum size for a full node is 300G and 100G for a pruned one", which will probably be relevant for a few years? WDYT?

Reply:
> I've never tried that though, is simply adding --prune-blockchain to the daemon start params enough?

It will prune the future blocks, if you want to prune previously downloaded blockchain you have to use pruning tool

Reply:
It will prune all* blocks*, but wont shrink the lmbd file size

Reply:
@nahuhh sorry, thanks for correction.